File: src/test/java/com/pschlup/ta/indicators/LowestValueTest.java
package com.pschlup.ta.indicators;

import com.pschlup.ta.helpers.MockTimeSeries;
import org.junit.jupiter.api.Test;

import static com.google.common.truth.Truth.assertThat;

class LowestValueTest {

  private double[] PRICES = new double[] {
      6, 5, 4, 22, 2, 1, 0, -30, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,
  };

  @Test
  void shouldComputeLowestValue() {
    Indicator lowest = LowestValue.of(MockTimeSeries.of(PRICES).closePrice()).withLength(15);
    assertThat(lowest.valueAt(0)).isEqualTo(-30);
  }
}

File: src/test/kotlin/com/pschlup/ta/indicators/LowestValueTest.kt
package com.pschlup.ta.indicators

import com.google.common.truth.Truth.assertThat
import com.pschlup.ta.helpers.FakeTimeSeries
import org.junit.jupiter.api.Test

internal class LowestValueTest {
  private val timeSeries = FakeTimeSeries.of(
    doubleArrayOf(6.0, 5.0, 4.0, 22.0, 2.0, 1.0, 0.0, -30.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0)
  )

  @Test
  fun shouldComputeLowestValue() {
    val lowest: Indicator = timeSeries.closePrice.lowestValue(15)

    assertThat(lowest.valueAt(0)).isEqualTo(-30)
  }
}
